ncm-dirperm/src/main/perl/dirperm.pm

```diff
@@ -155,64 +175,85 @@ sub process_path {
     my $correct_ownership = 0;
     my $correct_permissions = 0;
     # Check if file exists and if permissions are correct
-	if ( -e $path ) {
-		$self->debug(1,"$path exists");
-		$exists = 1;
-		my ($_dev,$_ino,$_mode,$_nlink,$_uid,$_gid,$_rdev,$_size,
-				$_atime,$_mtime,$_ctime,$_blksize,$_blocks) = stat($path);
-		# Show permissions in conventional octal format
-		my $_operm = sprintf "%04o", S_IMODE($_mode);
-		my $operm = sprintf "%04o", $perm;
-
-		if(($uid == $_uid) and ($gid == $_gid)){
-			$self->debug(1,"$path has correct ownership $_uid:$_gid");
-			$correct_ownership = 1;
-		} else {
-			$self->debug(1,"$path has ownership $_uid:$_gid, not $uid:$gid");
-		};
-
-		if($operm == $_operm){
-			$self->debug(1,"$path has correct permissions $_operm" );
-			$correct_permissions = 1;
-		} else {
-			$self->debug(1,"$path has permissions $_operm, not $operm");
-		}
-	}
+    if ( -e $path ) {
+        $self->debug(1,"$path exists");
+        $exists = 1;
+        my ($_dev,$_ino,$_mode,$_nlink,$_uid,$_gid,$_rdev,$_size,
+                $_atime,$_mtime,$_ctime,$_blksize,$_blocks) = stat($path);
+        # Show permissions in conventional octal format
+        my $_operm = sprintf "%04o", S_IMODE($_mode);
+        my $operm = sprintf "%04o", $perm;
+
+        if(($uid == $_uid) and ($gid == $_gid)){
+            $self->debug(1,"$path has correct ownership $_uid:$_gid");
+            $correct_ownership = 1;
+        } else {
+            $self->debug(1,"$path has ownership $_uid:$_gid, not $uid:$gid");
+        };
+
+        if($operm == $_operm){
+            $self->debug(1,"$path has correct permissions $_operm" );
+            $correct_permissions = 1;
+        } else {
+            $self->debug(1,"$path has permissions $_operm, not $operm");
+        }
+    }
     # Make the file or directory.
-	if(not $exists) {
-		if ($type eq "f") {
-			# Make the file.
-			open TMP, ">>$path";
-			close TMP;
-			if (! -f $path) {
-				$self->error("error creating file $path");
-				return 0;
-			}
-		} elsif ($type eq "d") {
-			# Make the directory and any parent directories.
-			eval { mkpath($path,0,$perm) };
-			if ($@) {
-				$self->error("error making directory: $@");
-				return 0;
-			}
-		}
-	}
-
-	if(not $correct_ownership) {
-		my $cnt = chown $uid, $gid, $path;
-		if ($cnt != 1) {
-			$self->error("can't change owner on $path");
-			return 0;
-		}
-	}
-
-	if(not $correct_permissions) {
-		my $cnt = chmod $perm, $path;
-		if ($cnt != 1) {
-			$self->error("can't change permissions on $path");
-			return 0;
-		}
-	}
+    if(not $exists) {
+        if ($type eq "f") {
+            # Make the file.
+            open TMP, ">>$path";
```

Comment:
what are you trying to achieve here? it creates a file when none exists, but of one exists, you don't actully do anything (except opening it). If that is ok, you can use FileEditor for this.
```perl
CAF::FileEditor->new($path, log=>$self)->close();                                                                                                        
if($EC->error()) {                                                                                                                                    
    $self->error("Error creating file $path:".$EC->error()->reason());
    $EC->ignore_error();
    return 0;
}
```